File: 2020/3.py
trees = []
with open("3.txt") as file:
    for line in file:
        trees.append(line.strip()*10000)

down = 1
right = 3
count = 0

while True:
    if trees[down][right] == "#":
        count += 1
    if down == len(trees) - 1:
        break
    down += 1
    right += 3
print(count)

slopes = [[1, 1], [1, 3], [1, 5], [1, 7], [2, 1]] #down, right
treeCount = []
for slope in slopes:
    count = 0
    down = slope[0]
    right = slope[1]
    while True:
        if trees[down][right] == "#":
            count += 1
        if down == len(trees) - 1:
            treeCount.append(count)
            break
        down += slope[0]
        right += slope[1]

count = 1
for trees in treeCount:
    count = count * trees
print(count)

File: 2020/Lua/3.lua
local grid = {}
local slopes = {{1,1}, {3,1}, {5,1}, {7,1}, {1,2}}
for line in io.lines("3.txt") do
  table.insert(grid, line)
end

local scores = {}
for n, coords in pairs(slopes) do
  local x, y, count = 1, 1, 0
  while true do
    if not grid[y] then break end
    if grid[y]:sub(x % #grid[y] ~= 0 and x % #grid[y] or #grid[y], x % #grid[y] ~= 0 and x % #grid[y] or #grid[y]) == "#" then
      count = count + 1
    end
    x = x + coords[1]
    y = y + coords[2]
  end
  if n == 2 then
    print(count)
  end
  table.insert(scores, count)
end

local prod = 1
for _, v in pairs(scores) do
  prod = prod * v
end
print(prod)
